service/history/task/transfer_active_task_executor.go

```diff
@@ -630,6 +635,25 @@ func (t *transferActiveTaskExecutor) processStartChildExecution(
 		return err
 	}
 
+	// Get target domain name
+	var targetDomainName string
+	var targetDomainEntry *cache.DomainCacheEntry
+	if targetDomainEntry, err = t.shard.GetDomainCache().GetDomainByID(task.TargetDomainID); err != nil {
+		if _, ok := err.(*types.EntityNotExistsError); !ok {
+			return err
+		}
+		// TODO: handle the case where target domain does not exist
+
+		// it is possible that the domain got deleted. Use domainID instead as this is only needed for the history event
+		targetDomainName = task.TargetDomainID
+	} else {
+		if targetCluster := targetDomainEntry.GetReplicationConfig().ActiveClusterName; targetCluster != t.shard.GetClusterMetadata().GetCurrentClusterName() {
```

Comment:
move to a function isCrossClusterTask(activeClusterName string) bool ?

Reply:
Sure, will do.

Reply:
Updated